qmlbind/include/qmlbind/metaclass.h

```diff
@@ -98,17 +96,39 @@ QMLBIND_API void qmlbind_exporter_add_signal(
  * `qmlbind_client_callbacks`' `read_property` & `write_property` functions.
  *
  * \param notifierSignal has to be a valid signal name, which means neither a nullptr nor an empty string. The
- * signal should be added with `qmlbind_exporter_add_signal` in advance, or else a warning will be issued.
+ * signal should be added with `qmlbind_metaclass_add_signal` in advance, or else a warning will be issued.
  * Notify signals are usually named `<propertyname>Changed` in Qt.
  *
  * libqmlbind's equivalent of [Q_PROPERTY](https://doc.qt.io/qt-5/properties.html), always using READ, WRITE and NOTIFY.
  */
-QMLBIND_API void qmlbind_exporter_add_property(
-    qmlbind_exporter *self,
+QMLBIND_API void qmlbind_metaclass_add_property(
+    qmlbind_metaclass *self,
     const char *name,
     const char *notifierSignal
 );
 
+/*!
+ * \brief exposes the `metaclass` as `qmlName` component to QML.
+ *
+ * `uri`, `versionMajor` and `versionMinor` is the package uri and version under which this `metaclass` is made
+ * available to QML. After calling this function, you can import the component as:
+ *
+ * ```
+ * import <uri> <versionMajor>.<versionMinor>
+ * ```
+ *
+ * After registering, metaclass changes such as method addition will not be reflected in the QML type system.
+ *
```

Comment:
Let's add a note here about the required lifetime of the passed metaclass? As I see it, a client could destoy the metaclass directly after registering if he wants to, but should keep it around in case he wants to use `qmlbind_engine_new_wrapper` later on.